client/src/modules/Opportunities/components/EditCv/ContactsForm/index.test.tsx

```diff
@@ -0,0 +1,74 @@
+  phone: '+1111111111111',
+  skype: 'some_skype',
+  telegram: 'some_telegram',
+  website: 'https://example.com',
+};
+
+describe('ContactsForm', () => {
+  test('form fields should have proper values', async () => {
+    render(<ContactsForm contactsList={mockContactsList} />);
+
+    const email = await screen.findByDisplayValue(mockContactsList.email);
+    const githubUsername = await screen.findByDisplayValue(mockContactsList.githubUsername);
+    const linkedin = await screen.findByDisplayValue(mockContactsList.linkedin);
+    const phone = await screen.findByDisplayValue(mockContactsList.phone);
+    const skype = await screen.findByDisplayValue(mockContactsList.skype);
+    const telegram = await screen.findByDisplayValue(mockContactsList.telegram);
+    const website = await screen.findByDisplayValue(mockContactsList.website);
+
+    expect(email).toBeInTheDocument();
+    expect(githubUsername).toBeInTheDocument();
+    expect(linkedin).toBeInTheDocument();
+    expect(phone).toBeInTheDocument();
+    expect(skype).toBeInTheDocument();
+    expect(telegram).toBeInTheDocument();
+    expect(website).toBeInTheDocument();
+  });
+
+  test('form fields should have proper placeholders', async () => {
+    render(<ContactsForm contactsList={mockContactsList} />);
+
+    const email = await screen.findByPlaceholderText('Email');
+    const githubUsername = await screen.findByPlaceholderText('Github username');
+    const linkedin = await screen.findByPlaceholderText('LinkedIn username');
+    const phone = await screen.findByPlaceholderText('+12025550111');
+    const skype = await screen.findByPlaceholderText('Skype id');
+    const telegram = await screen.findByPlaceholderText('Telegram public name');
+    const website = await screen.findByPlaceholderText('Enter your website URL');
+
+    expect(email).toBeInTheDocument();
+    expect(githubUsername).toBeInTheDocument();
+    expect(linkedin).toBeInTheDocument();
+    expect(phone).toBeInTheDocument();
+    expect(skype).toBeInTheDocument();
+    expect(telegram).toBeInTheDocument();
+    expect(website).toBeInTheDocument();
+  });
+
+  test('form fields should have proper labels', async () => {
+    render(<ContactsForm contactsList={mockContactsList} />);
+
+    const email = await screen.findByLabelText('Email');
+    const githubUsername = await screen.findByLabelText('Github');
+    const linkedin = await screen.findByLabelText('LinkedIn');
+    const phone = await screen.findByLabelText('Phone');
+    const skype = await screen.findByLabelText('Skype');
+    const telegram = await screen.findByLabelText('Telegram');
+    const website = await screen.findByLabelText('Website');
+
+    expect(email).toBeInTheDocument();
+    expect(githubUsername).toBeInTheDocument();
+    expect(linkedin).toBeInTheDocument();
+    expect(phone).toBeInTheDocument();
+    expect(skype).toBeInTheDocument();
+    expect(telegram).toBeInTheDocument();
+    expect(website).toBeInTheDocument();
+  });
+});
```

Comment:
```
describe('ContactsForm', () => {
  test.each`
    value                              | placeholder                 | labelText
    ${mockContactsList.email}          | ${'Email'}                  | ${'Email'}
    ${mockContactsList.githubUsername} | ${'Github username'}        | ${'Github'}
    ${mockContactsList.linkedin}       | ${'LinkedIn username'}      | ${'LinkedIn'}
    ${mockContactsList.phone}          | ${'+12025550111'}           | ${'Phone'}
    ${mockContactsList.skype}          | ${'Skype id'}               | ${'Skype'}
    ${mockContactsList.telegram}       | ${'Telegram public name'}   | ${'Telegram'}
    ${mockContactsList.website}        | ${'Enter your website URL'} | ${'Website'}
  `('form field should have proper value, placeholder and label', async ({ value, placeholder, labelText }) => {
    render(<ContactsForm contactsList={mockContactsList} />);
    const el = await screen.findByDisplayValue(value);
    expect(el).toBeInTheDocument();
    const elByPlaceholder = await screen.findByPlaceholderText(placeholder);
    expect(elByPlaceholder).toBeInTheDocument();
    const label = await screen.findByLabelText(labelText);
    expect(label).toBeInTheDocument();
  });
});
```

maybe that would be better?

Reply:
I thought about table tests here, but I would like to keep these tests separated + it requires excessive renders for each generated case, I would like to avoid it

Reply:
Agree with @valerydluski. It's better to use tests table and separate each test. Additional renders do not cost here so much, but when your tests will fails it will be harder to understand which line was broken. When it will have each separate test for field with field name it will be more cleaner to read and understand which case exactly has failed.
Also, it's a best practice to have more specific test and separate each `expect` into it's own test

Reply:
fixed